test/factories/feature.rb

```diff
@@ -1,5 +1,7 @@
 FactoryBot.define do
   factory :feature do
+    initialize_with { Feature.find_or_create_by(name: name)}
```

Comment:
I did find that the Feature model  is in no way unique and since the code attempts to find the Feature by name, it returns the first entry while the factory uses an ID of the just created instance. This works around that

Reply:
https://github.com/theforeman/foreman/pull/6992 is my stab at improving this.